Makefile

```diff
@@ -196,8 +196,18 @@ else
 endif
 
 gobgp:
-	cd vendor/github.com/osrg/gobgp/gobgp && \
-	CGO_ENABLED=0 GOARCH=$(GOARCH) GOOS=linux go build -o $(MAKEFILE_DIR)/gobgp
+ifeq "$(BUILD_IN_DOCKER)" "true"
+	@echo Building gobgp
+	$(DOCKER) run -v $(PWD)/vendor:/go/src -w /go/src/github.com/osrg/gobgp/gobgp $(DOCKER_BUILD_IMAGE) \
+    sh -c ' \
+	    GOARCH=$(GOARCH) CGO_ENABLED=0 go build -o gobgp'
+	@echo Finished building gobgp.
+else
+	pushd vendor/github.com/osrg/gobgp/gobgp && \
```

Comment:
Do we need `pushd`? Why do we need push and pop here?

```sh
± make gobgp                                                                                                                                                                                                                                                                          ❯❯❯
Building for GOARCH=amd64
pushd vendor/github.com/osrg/gobgp/gobgp && \
CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -o gobgp
/bin/sh: 1: pushd: not found
Makefile:206: recipe for target 'gobgp' failed
```

with build in docker set to true I am running in to this error.

```
± make gobgp                                                                                                                                                                                                                                                                          ❯❯❯
Building for GOARCH=amd64
Building gobgp
docker run -v /home/murali/Desktop/projects/go/src/github.com/cloudnativelabs/kube-router/vendor:/go/src -w /go/src/github.com/osrg/gobgp/gobgp golang:1.10.8-alpine3.9 \
    sh -c ' \
    GOARCH=amd64 CGO_ENABLED=0 go build -o gobgp'
Finished building gobgp.
ln -f vendor/github.com/osrg/gobgp/gobgp/gobgp gobgp
ln: failed to create hard link 'gobgp' => 'vendor/github.com/osrg/gobgp/gobgp/gobgp': Operation not permitted
Makefile:200: recipe for target 'gobgp' failed
```

Could you please take a look?

Reply:
ouch... you're right; pushd/popd aren't needed as each target rule is executed in a standalone shell instance.

Thanks for testing!